packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -435,32 +491,24 @@ export abstract class AbstractVSXExtensionComponent<Props extends AbstractVSXExt
     };
 
     protected renderAction(host?: TreeWidget): React.ReactNode {
-        const { builtin, busy, plugin } = this.props.extension;
+        const { builtin, currentTask, disabled, uninstalled, installed, deployed } = this.props.extension;
         const isFocused = (host?.model.getFocusedNode() as TreeElementNode)?.element === this.props.extension;
         const tabIndex = (!host || isFocused) ? 0 : undefined;
-        const installed = !!plugin;
-        const outOfSynch = plugin?.metadata.outOfSync;
-        if (builtin) {
-            return <div className="codicon codicon-settings-gear action" tabIndex={tabIndex} onClick={this.manage}></div>;
+        const outOfSync = installed && (deployed ? (disabled || uninstalled) : !(disabled || uninstalled));
+        if (currentTask) {
+            return <button className="theia-button action prominent theia-mod-disabled">{currentTask}</button>;
         }
-        if (busy) {
-            if (installed) {
-                return <button className="theia-button action theia-mod-disabled">{nls.localizeByDefault('Uninstalling')}</button>;
+        return <div>
+            {
+                outOfSync && <button className="theia-button action" onClick={this.reloadWindow}>{nls.localizeByDefault('Reload Window')}</button>
             }
-            return <button className="theia-button action prominent theia-mod-disabled">{nls.localizeByDefault('Installing')}</button>;
-        }
-        if (installed) {
-            return <div>
-                {
-                    outOfSynch
-                        ? <button className="theia-button action" onClick={this.reloadWindow}>{nls.localizeByDefault('Reload Window')}</button>
-                        : <button className="theia-button action" onClick={this.uninstall}>{nls.localizeByDefault('Uninstall')}</button>
-                }
-
-                <div className="codicon codicon-settings-gear action" onClick={this.manage}></div>
-            </div>;
-        }
-        return <button className="theia-button prominent action" onClick={this.install}>{nls.localizeByDefault('Install')}</button>;
+            {
+                !builtin && ((installed && !uninstalled) ?
+                    <button className="theia-button action" onClick={this.uninstall}>{nls.localizeByDefault('Uninstall')}</button> :
+                    <button className="theia-button prominent action" onClick={this.install}>{nls.localizeByDefault('Install')}</button>)
+            }
+            <div className="codicon codicon-settings-gear action" tabIndex={tabIndex} onClick={this.manage}></div>;
```

Comment:
Bug: There is an additional `;` which gets rendered into the HTML:

![image](https://github.com/user-attachments/assets/72494a1b-ba98-45cd-96f6-7170b27773a3)

```suggestion
            <div className="codicon codicon-settings-gear action" tabIndex={tabIndex} onClick={this.manage}></div>
```